core/src/components/accordion/accordion.ts

```diff
@@ -589,122 +543,162 @@ function createAccordionItem(
+): WidgetFactory<AccordionWidget> {
+	return (config?: PropsConfig<AccordionProps>) => {
+		const [
+			{
+				closeOthers$,
+				onShown$,
+				onHidden$,
+				className$,
+				itemId$,
+				itemAnimated$,
+				itemClass$,
+				itemDisabled$,
+				itemVisible$,
+				itemTransition$,
+				itemDestroyOnHide$,
+				itemBodyClass$,
+				itemButtonClass$,
+				itemBodyContainerClass$,
+				itemHeaderClass$,
+				itemHeadingTag$,
+				onItemVisibleChange$,
+				onItemHidden$,
+				onItemShown$,
+				...stateProps
+			},
+			patch,
+		] = writablesForProps(defaultAccordionConfig, config, configAccordionValidator);
+		const accordionItemConfig: ReadableSignals<AccordionItemProps> = {
+			itemId: itemId$,
+			itemClass: itemClass$,
+			itemAnimated: itemAnimated$,
+			itemDisabled: itemDisabled$,
+			itemVisible: itemVisible$,
+			itemTransition: itemTransition$,
+			itemDestroyOnHide: itemDestroyOnHide$,
+			itemBodyClass: itemBodyClass$,
+			itemButtonClass: itemButtonClass$,
+			itemBodyContainerClass: itemBodyContainerClass$,
+			itemHeaderClass: itemHeaderClass$,
+			itemHeadingTag: itemHeadingTag$,
+			onItemVisibleChange: onItemVisibleChange$,
+			onItemHidden: onItemHidden$,
+			onItemShown: onItemShown$,
+		};
+		const itemsWidget$ = registrationArray<AccordionItemWidget>();
+		const openItems$ = computed(() => {
+			const openItems: string[] = [];
+			itemsWidget$().forEach((item) => {
+				if (item.state$().itemVisible) {
+					openItems.push(item.state$().itemId);
+				}
+			});
+			return openItems;
+		});
+
+		const oldOpenItem$ = writable(openItems$()[0]);
+		const checkCloseOthersAction$ = computed(() => {
+			if (closeOthers$()) {
+				adjustItemsCloseOthers(itemsWidget$(), openItems$(), oldOpenItem$());
+				oldOpenItem$.set(openItems$()[0]);
 			}
 		});
-		return openItems;
-	});
 
-	const oldOpenItem$ = writable(openItems$()[0]);
-	const checkCloseOthersAction$ = computed(() => {
-		if (closeOthers$()) {
-			adjustItemsCloseOthers(itemsWidget$(), openItems$(), oldOpenItem$());
-			oldOpenItem$.set(openItems$()[0]);
-		}
-	});
+		const action$ = computed(() => {
+			checkCloseOthersAction$();
+		});
+		return {
+			...stateStores({itemsWidget$, className$, ...stateProps}),
+			patch,
+			actions: {},
+			api: {
+				expand: (id: string) => {
+					getItem(itemsWidget$(), id)?.api.expand();
+				},
+				collapse: (id: string) => {
+					getItem(itemsWidget$(), id)?.api.collapse();
+				},
+				toggle: (id: string) => {
+					getItem(itemsWidget$(), id)?.api.toggle();
+				},
+				expandAll: () => {
+					itemsWidget$().forEach((i) => i.api.expand());
+				},
+				collapseAll: () => {
+					itemsWidget$().forEach((i) => i.api.collapse());
+				},
+				registerItem: (propsConfig?: PropsConfig<AccordionItemProps>) => {
+					const itemProps = accordionItemProps as (keyof AccordionItemProps)[];
+					const config = mergeConfigStores(itemProps, normalizeConfigStores(itemProps, propsConfig?.config), accordionItemConfig);
```

Comment:
As `accordionItemConfig` does not include the extra props (that can be mentioned in `itemProps`), I am wondering if it actually works to specify a slot at the accordion level that will be used by default by its child accordion items.

Reply:
Indeed, I have updated this PR with a new commit (easier to review, seeing the size of this PR) that handles this.
I also fixed stackblitz in local, which was broken.